pkg/network/go/binversion/buildinfo.go

```diff
@@ -49,6 +51,20 @@ var (
 	buildInfoMagic = []byte("\xff Go buildinf:")
 )
 
+const (
+	buildInfoAlign = 16
+	buildInfoSize  = 32
+	maxSizeToRead  = 64 * 1024 // 64KB
+)
+
+var (
+	dataRawBuildPool = sync.Pool{
+		New: func() any {
+			return make([]byte, maxSizeToRead)
+		},
+	}
+)
+
```

Comment:
This will actually result in an allocation (not because of the slice data itself, which will be correctly pooled, but because of the `[]byte` to `any` interface conversion when you call `Put()`)
To solve this you need to return a _pointer_ to the slice (it's one of the few instances where this actually makes sense 😅 )

Unfortunately there is no direct mention of this in the documentation, although there is a subtle note alluding to this buried deep down in the examples https://pkg.go.dev/sync#example-Pool

Reply:
thanks, made the changes